File: src/main/java/com/runehive/game/event/impl/PickupItemEvent.java
package com.runehive.game.event.impl;

import com.runehive.game.event.Event;
import com.runehive.game.world.items.Item;
import com.runehive.game.world.items.ground.GroundItem;
import com.runehive.game.world.position.Position;

public class PickupItemEvent implements Event {

    private final GroundItem groundItem;

    public PickupItemEvent(GroundItem groundItem) {
        this.groundItem = groundItem;
    }

    public GroundItem getGroundItem() {
        return groundItem;
    }

    public Item getItem() {
        return groundItem.item;
    }

    public Position getPosition() {
        return groundItem.getPosition();
    }

}

File: src/main/kotlin/org/jire/runehiveps/event/item/PickupItemEvent.kt
package org.jire.runehiveps.event.item

import com.runehive.content.event.EventDispatcher
import com.runehive.content.event.impl.PickupItemInteractionEvent
import com.runehive.game.world.entity.mob.player.Player
import com.runehive.game.world.entity.mob.player.PlayerRight
import com.runehive.game.world.items.Item
import com.runehive.game.world.position.Position
import com.runehive.net.packet.out.SendMessage

/**
 * @author Jire
 */
class PickupItemEvent(
    val id: Int,
    val x: Int, val y: Int
) : ItemEvent {

    override fun handle(player: Player) {
        val item = Item(id)
        val position = Position.create(x, y, player.height)

        if (EventDispatcher.execute(player, PickupItemInteractionEvent(item, position))) {
            if (PlayerRight.isDeveloper(player)) {
                player.send(
                    SendMessage(
                        String.format(
                            "[%s]: item=%d position=%s",
                            PickupItemInteractionEvent::class.java.simpleName, item.id, position.toString()
                        )
                    )
                )
            }
            return
        }

        player.pickup(item, position)
    }

}